nvdaHelper/remote/sysListView32.cpp

```diff
@@ -59,3 +59,46 @@ error_status_t nvdaInProcUtils_sysListView32_getColumnContent(handle_t bindingHa
 	*text = SysAllocString(lvItem.pszText);
 	return 0;
 }
+
+error_status_t nvdaInProcUtils_sysListView32_getColumnLocation(handle_t bindingHandle, const unsigned long windowHandle, int item, int subItem, RECT* location) {
+	RECT localRect {
+		// Returns the bounding rectangle of the entire item, including the icon and label.
+		LVIR_LABEL,  // left
+		// According to Microsoft, top should be the one-based index of the subitem.
+		// However, indexes coming from LVM_GETCOLUMNORDERARRAY are zero based.
+		subItem // top
+	};
+	if (!SendMessage((HWND)UlongToHandle(windowHandle), LVM_GETSUBITEMRECT, (WPARAM)item, (LPARAM)&localRect)) {
+		LOG_DEBUGWARNING(L"LVM_GETSUBITEMRECT failed");
+		return 1;
+	}
+	*location = localRect;
+	return 0;
+}
+
+error_status_t nvdaInProcUtils_sysListView32_getColumnHeader(handle_t bindingHandle, const unsigned long windowHandle, int subItem, BSTR* text) {
+	LVCOLUMN lvColumn = {0};
+	lvColumn.mask = LVCF_TEXT;
+	lvColumn.iSubItem = subItem;
+	lvColumn.cchTextMax = CBEMAXSTRLEN;
+	wchar_t textBuf[CBEMAXSTRLEN]{}; // Ensure that the array initialised with all zero values ('\0')
+	lvColumn.pszText= textBuf;
+	if (!SendMessage((HWND)UlongToHandle(windowHandle), LVM_GETCOLUMN, (WPARAM)subItem, (LPARAM)&lvColumn)) {

```

Comment:
The use of SendMessage surprises me, being an in process call. I guess there is no explicit API for this. It would be good to have some developer docs about this.

Reply:
There are macro's, see for example: https://docs.microsoft.com/en-us/windows/win32/api/commctrl/nf-commctrl-listview_getsubitemrect

While Microsoft recommends these macro's, they return void, which is really bad IMO.

Reply:
Note that the [ListView_GetColumn macro](https://docs.microsoft.com/en-us/windows/win32/controls/lvm-getcolumn) does have a return value.

For any others that don't, please add a comment. An example comment for if the `ListView_GetColumn` macro had no return value, yet the `LVM_GETCOLUMN` constant declared one:
``` C++
// ListView_GetColumn macro has no return value, using SendMessage directly so errors are caught.
// The meaning of the return value depends on the message sent, for LVM_GETCOLUMN ,
// it returns TRUE if successful, or FALSE otherwise.
```

Reply:
I'm confused. [Documentation here](https://docs.microsoft.com/en-us/windows/desktop/api/Commctrl/nf-commctrl-listview_getcolumn) states void as return value.

Reply:
Yes, you're right, I must have misread that.